snap/hooks/post-refresh

```diff
@@ -0,0 +1,43 @@
+#!/bin/sh
+set -eu
+
+PREVIOUS_VERSION=$(snapctl get previous-version)
+INITIAL_ALLOWED_USERS_VERSION="0.2.0"
+
+version_less_than() {
+  [ "$(printf '%s\n' "${@}" | sort -V | head -n1)" = "$1" ]
+}
+
+echo_and_log_to_journal() {
+  echo "${@}"
+  logger -t $SNAP_NAME "${@}"
+}
+
+should_transition_to_allowed_users() {
+  # Transition to allowed users if:
+  # - previous-version is not set (that means that the previous version is
+  #   older than 0.2.0, i.e. the version where we introduced setting the
+  #   previous-version in the pre-refresh hook).
+  # - previous-version is set, but it is less than 0.2.0. That should never
+  #   happen, but we check it to give an example how the previous-version
+  #   can be used to transition data from older versions.
+  [ -z "${PREVIOUS_VERSION}" ] || version_less_than "${PREVIOUS_VERSION}" "${INITIAL_ALLOWED_USERS_VERSION}"
+}
+
+transition_to_allowed_users() {
+  echo_and_log_to_journal "Transitioning to allowed users"
+  mkdir -p ${SNAP_DATA}/broker.conf.d
```

Comment:
```suggestion
  mkdir -p --mode=700 ${SNAP_DATA}/broker.conf.d
```

Reply:
I noticed that the `broker.conf` is created with the default permissions in https://github.com/nsklikas/authd-oidc-brokers/blob/4e490fdbeabb7c6b2331493775bbe924ccdbacc1/snap/hooks/install#L4. IMO we should be consistent, so either we create both with default permissions or with 700. I'm leaning towards the latter, because the config file can contains secrets (e.g. the `client_secret`).

Reply:
See https://github.com/ubuntu/authd-oidc-brokers/pull/262